docs/struct_constructors.md

```diff
@@ -0,0 +1,118 @@
+Struct constructors
+===================
+
+This documents various types of constructors that can be defined for structs in LIME IDL.
+
+Implicit constructors
```

Comment:
I would add that implicit and field constructors are generated in specific language only without crossing languages boundaries.
Also despite that it's quite obvious for implicit constructors I would specify order in which parameters appear in constructor

Reply:
Updated